packages/invoices/src/InboundShipment/DetailView/DetailView.tsx

```diff
@@ -1,60 +1,108 @@
   getRowExpandColumn,
   useDialog,
   DialogButton,
   useTranslation,
-  Item,
+  useMutation,
+  useQueryClient,
 } from '@openmsupply-client/common';
 
-import { InboundAction, reducer } from './reducer';
 import { getInboundShipmentDetailViewApi } from './api';
 import { Toolbar } from './Toolbar';
 import { Footer } from './Footer';
 import { AppBarButtons } from './AppBarButtons';
 import { SidePanel } from './SidePanel';
 import { GeneralTab } from './GeneralTab';
 import { InboundLineEdit } from './modals/InboundLineEdit/InboundLineEdit';
-
-import { isInboundEditable } from '../../utils';
-import { InboundShipmentItem } from '../../types';
+import { inboundLinesToSummaryItems } from './reducer/reducer';
+import {
+  getNextInboundStatus,
+  isInboundEditable,
+  placeholderInbound,
+} from '../../utils';
+import {
+  InboundShipmentItem,
+  InboundShipment,
+  Invoice,
+  OutboundShipmentSummaryItem,
+} from '../../types';
 
 const useDraftInbound = () => {
+  const queryClient = useQueryClient();
   const { id } = useParams();
   const { api } = useOmSupplyApi();
+  const queries = getInboundShipmentDetailViewApi(api);
 
-  const { draft, save, dispatch, state } = useDocument(
-    ['invoice', id],
-    reducer,
-    getInboundShipmentDetailViewApi(api)
-  );
+  const onChangeSortBy = () => {};
+
+  const { data } = useQuery(['invoice', id], () => {
+    return queries.onRead(id);
+  });
+
+  const draft = data
+    ? { ...data, items: inboundLinesToSummaryItems(data?.lines ?? []) }
+    : placeholderInbound;
+
+  const { mutateAsync } = useMutation(queries.onUpdate, {
+    onMutate: async (patch: Partial<InboundShipment>) => {
+      await queryClient.cancelQueries(['invoice', id]);
+
+      const previousInbound: Invoice = queryClient.getQueryData([
+        'invoice',
+        id,
+      ]);
+
+      queryClient.setQueryData(['invoice', id], {
+        ...previousInbound,
+        ...patch,
+      });
+
+      return { previousInbound, patch };
+    },
+    onSettled: () => queryClient.invalidateQueries(['invoice', id]),
+    onError: (_, __, context) => {
+      queryClient.setQueryData(['invoice', id], context.previousInbound);
+    },
+  });
+
+  const { isLoading: isAddingItem, mutateAsync: noOptimisticMutate } =
+    useMutation(queries.onUpdate, {
+      onSettled: () => queryClient.invalidateQueries(['invoice', id]),
+    });
 
-  const onChangeSortBy = (column: Column<InboundShipmentItem>) => {
-    dispatch(InboundAction.onSortBy(column));
+  const updateInvoice = async (patch: Partial<InboundShipment>) => {
+    return mutateAsync({ ...data, ...patch, items: [] });
   };
 
-  return { draft, save, dispatch, onChangeSortBy, sortBy: state.sortBy };
-};
+  const upsertItem = async (item: OutboundShipmentSummaryItem) => {
+    const itemIdx = draft.items.findIndex(i => i.id === item.id);
+    if (itemIdx >= 0) draft.items[itemIdx] = item;
+    else draft.items.push(item);
+
+    // throw new Error('testing!');
```

Comment:
was just using this for to test the behaviour when throwing an error when adding/updating an item